### PR TITLE
New docs

### PR DESCRIPTION
**Pull request by reubenharry**
Mkdocs. Covers both docs and website. Looks nicer. Still on netlify.

### CODE REVIEW

**Comment by reubenharry** [comment]
@mknorps @idontgetoutmuch Here are what the new docs look like: https://monad-bayes.netlify.app/ The motivation was that currently I'm hosting two separate sites, one for the docs and one for the notebooks, and this both looks cleaner and is easier to maintain. I think this repo is set up such that I can't merge this in myself, so if one of you could, that would be great, and then I can delete the old websites.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: ci 
+on:
+  push:
+    branches:
+      - newdocs
```
Comment (mknorps) [inline .github/workflows/ci.yml:5]: Change trigger to push on master and add cache.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+      - run: pip install mkdocs-material 
+      - run: mkdocs gh-deploy --force
```
Comment (mknorps) [inline .github/workflows/ci.yml]: Why do we need `--force` here?

File: mkdocs.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
+site_name: Probabilistic Programming in Haskell
+theme:
+  name: material
+  features:
+    # - announce.dismiss
```
Comment (mknorps) [inline mkdocs.yml]: Are the commented features still needed? If not, please remove.

File: mkdocs.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    - navigation.sections
+    # - navigation.tabs
+    # - navigation.tabs.sticky
+    - navigation.top
+    - navigation.tracking
+    - search.highlight
+    - search.share
+    - search.suggest
+    - toc.follow
+  palette:
```
Comment (mknorps) [inline mkdocs.yml]: Nice! I wonder if this will work with bright and dark mode of the users' browsers.

File: runtime.txt [hunk 5]
```diff
@@ -0,0 +1 @@
+3.8
```
Comment (mknorps) [inline runtime.txt]: It would be better to use Python 3.10 or 3.11. The maintanance period for them will be longer.

File: docs/usage.md [hunk 6]
```diff
@@ -654,23 +654,6 @@ model = do
 
 is really an unnormalized measure, rather than a probability distribution. `normalize` views it as of type `Weighted Integrator Double`, which is isomorphic to `(Double -> (Double, Log Double) -> Double)`. This can be used to compute the normalization constant, and divide the integrator's output by it, all within `Integrator`.
```
Comment (mknorps) [inline docs/usage.md]: Why you do not want to include "Quadrature" in this version of the documentation?

**Review by mknorps — CHANGES_REQUESTED** [review]
The site looks elegant! It has better user experience and you modernized it well. I like it very much.

Apart from the comments within the code, I have few comments and suggested changes regarding the PR:
- in `docs/images` plot.pgn and plot copy.pgn are duplicates. Remove the copy.
- How does `tintin` work here? I see a `.tintin.yml` file but no reference to it in teh code and requirements.
- Do we still need `monad-bayes-site` folder?
- For the structure of the project it would be better if the code to generate documentation was in a separate folder. 
- is `monad-bayes-site.cabal` and `stack.yaml` within `docs` still needed?

File: docs/usage.md (hunk 6 above)
Comment (reubenharry) [reply docs/usage.md]: It was duplicated.

File: runtime.txt (hunk 5 above)
Comment (reubenharry) [reply runtime.txt]: Done.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (reubenharry) [reply .github/workflows/ci.yml]: Removed.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (reubenharry) [reply .github/workflows/ci.yml:5]: Wasn't sure how to add cache, but added master.

**Comment by reubenharry** [comment]
For some reason, the --force and Python 3.8 were both necessary for CI and netlify to work respectively, but I changed the other things.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mknorps) [reply .github/workflows/ci.yml:5]: This should work:
```
      - name: Setup caching
        uses: actions/cache@v3
        with:
          key: ${{ github.sha }}
          path: .cache
```
Check out: https://gaseri.org/en/blog/2022-11-01-publishing-material-for-mkdocs-website-to-github-pages-using-custom-actions-workflow/

**Comment by mknorps** [comment]
> For some reason, the --force and Python 3.8 were both necessary for CI and netlify to work respectively, but I changed the other things.

That is inconvenient, but we can leave with this for now.

And what about:
> Do we still need monad-bayes-site folder?
> For the structure of the project it would be better if the code to generate documentation was in a separate folder.

Is also mathjax no longer needed?

**Comment by reubenharry** [comment]
> And what about:
> 
> > Do we still need monad-bayes-site folder?

I thought I removed it. I don't see it in the repo locally.

> > For the structure of the project it would be better if the code to generate documentation was in a separate folder.
> 

Now done.

> Is also mathjax no longer needed?

Re-added.

**Review by mknorps — CHANGES_REQUESTED** [review]
Thank you! The code structure looks better :)
One last thing - remove mkdocs.yaml from the main directory.

If you want to experiment a bit with caching, as I suggested in [this comment ](https://github.com/tweag/monad-bayes/pull/236/files#r1059271173) it would be great, but if you do not have time for this now, please create an issue for later not to forget about it.

**Comment by reubenharry** [comment]
OK, removed, and issue created.

**Comment by reubenharry** [comment]
Hm, netlify works fine, but something with CI fails. I'm a bit stumped. Any ideas?

**Comment by reubenharry** [comment]
Oh right, it's failing for the old site. Let me remove it...

**Comment by reubenharry** [comment]
Now passes CI, should be ready to merge.

**Review by mknorps — APPROVED** [review]
It looks OK now. Thank you for improving the documentation!

I created https://github.com/tweag/monad-bayes/issues/239 to take care of the developer's environment regarding developing documentation.

**Pull request merged**